### PR TITLE
feat: expose rpc server 

### PR DESCRIPTION
**Pull request by drspacemn**
Implement the starknet RPC spec:
https://github.com/starkware-libs/starknet-specs/tree/master/api

# Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing
- [ ] Other (please describe):

# What is the current behavior?

Issue Number: #29 

# What is the new behavior?

- declare and implement runtime APIs to expose the StarkNet pallet
- define a starknet JSON-RPC api
- implement the starknet JSON-RPC api

# Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by drspacemn** [comment]
Test RPC endpoints:
`curl -H "Content-Type: application/json" -d '{"id":1, "jsonrpc":"2.0", "method": "starknet_blockHashAndNumber"}' http://localhost:9933/`

`curl -H "Content-Type: application/json" -d '{"id":1, "jsonrpc":"2.0", "method": "starknet_blockNumber"}' http://localhost:9933/`

File: crates/pallets/starknet/src/lib.rs [hunk 1]
```diff
@@ -485,6 +489,16 @@ pub mod pallet {
 
     /// The Starknet pallet internal functions.
     impl<T: Config> Pallet<T> {
+        /// Get current block number.
+        ///
+        /// # Returns
+        ///
+        /// The current block number.
+        #[inline(always)]
+        pub fn current_block_number() -> Option<U256> {
```
Comment (EvolveArt) [inline crates/pallets/starknet/src/lib.rs:498]: I would rather stay with only `current_block` and do the mapping in the runtime `lib.rs`

File: crates/pallets/starknet/Cargo.toml [hunk 2]
```diff
@@ -30,6 +30,7 @@ frame-system = { workspace = true }
 frame-benchmarking = { workspace = true, optional = true }
 
 # Substrate primitives
+sp-api = { default-features = false, git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.40" }
```
Comment (EvolveArt) [inline crates/pallets/starknet/Cargo.toml:33]: we should probably add it to the parent `cargo.toml` like we do for the other crates

File: crates/pallets/starknet/src/lib.rs (hunk 1 above)
Comment (drspacemn) [reply crates/pallets/starknet/src/lib.rs]: moved to runtime lib

File: crates/pallets/starknet/Cargo.toml (hunk 2 above)
Comment (drspacemn) [reply crates/pallets/starknet/Cargo.toml]: moved to parent `cargo.toml`

**Review by EvolveArt — APPROVED** [review]
Did you check with @tdelabro ? If he is starting from this branch then let's merge it, lgtm

**Comment by tdelabro** [comment]
I use part of the code but in other files. So no need to merge. I took what was good and it will show up in my PR

**Pull request closed without merging**